scene/animation/animation_player.cpp

```diff
@@ -1240,11 +1294,173 @@ void AnimationPlayer::rename_animation(const StringName &p_name, const StringNam
+				// rename connections
+				animation_libraries[i].library->disconnect(SNAME("animation_added"), callable_mp(this, &AnimationPlayer::_animation_added));
+				animation_libraries[i].library->disconnect(SNAME("animation_removed"), callable_mp(this, &AnimationPlayer::_animation_added));
+				animation_libraries[i].library->disconnect(SNAME("animation_renamed"), callable_mp(this, &AnimationPlayer::_animation_renamed));
+
+				animation_libraries[i].library->connect(SNAME("animation_added"), callable_mp(this, &AnimationPlayer::_animation_added), varray(p_new_name));
+				animation_libraries[i].library->connect(SNAME("animation_removed"), callable_mp(this, &AnimationPlayer::_animation_added), varray(p_new_name));
+				animation_libraries[i].library->connect(SNAME("animation_renamed"), callable_mp(this, &AnimationPlayer::_animation_renamed), varray(p_new_name));
+
+				//rename library, if in use
+				if (animation_set.has(K.key) && animation_set[K.key].animation_library == p_name) {
+					animation_set[K.key].animation_library = p_new_name;
+				}
+			}
+		}
+	}
+
+	ERR_FAIL_COND(!found);
+
+	stop();
+
+	animation_libraries.sort(); // Must keep alphabetical order.
+
+	_animation_set_cache_update(); // Update cache.
+
+	notify_property_list_changed();
+}
+
+bool AnimationPlayer::has_animation_library(const StringName &p_name) const {
+	for (uint32_t i = 0; i < animation_libraries.size(); i++) {
+		if (animation_libraries[i].name == p_name) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+Ref<AnimationLibrary> AnimationPlayer::get_animation_library(const StringName &p_name) const {
+	for (uint32_t i = 0; i < animation_libraries.size(); i++) {
+		if (animation_libraries[i].name == p_name) {
+			return animation_libraries[i].library;
+		}
+	}
+	ERR_FAIL_V(Ref<AnimationLibrary>());
+}
+
+TypedArray<StringName> AnimationPlayer::_get_animation_library_list() const {
+	TypedArray<StringName> ret;
+	for (uint32_t i = 0; i < animation_libraries.size(); i++) {
+		ret.push_back(animation_libraries[i].name);
+	}
+	return ret;
+}
+
+void AnimationPlayer::get_animation_library_list(List<StringName> *p_libraries) const {
+	for (uint32_t i = 0; i < animation_libraries.size(); i++) {
+		p_libraries->push_back(animation_libraries[i].name);
+	}
+}
+
+TypedArray<String> AnimationPlayer::get_configuration_warnings() const {
+	TypedArray<String> warnings = Node::get_configuration_warnings();
+
+	for (uint32_t i = 0; i < animation_libraries.size(); i++) {
+		for (const KeyValue<StringName, Ref<Animation>> &K : animation_libraries[i].library->animations) {
+			if (animation_set.has(K.key) && animation_set[K.key].animation_library != animation_libraries[i].name) {
+				warnings.push_back(vformat(TTR("Animation '%s' in library '%s' is unused because another animation with the same name exists in library '%s'."), K.key, animation_libraries[i].name, animation_set[K.key].animation_library));
```

Comment:
Probably should be `RTR` instead of `TTR`.

Reply:
oh good point, thanks